tasks/DroneFlightControlTask.cpp

```diff
@@ -146,27 +139,38 @@ void DroneFlightControlTask::updateHook()
     States new_state = runtimeStatesTransition(control_device);
     if (state() != new_state)
     {
-        applyTransition(new_state);
+        if (new_state != CONTROLLING)
+        {
+            if(transitionToControlling(new_state))
+            {
+                state(CONTROLLING);
+            }
+            else
+            {
+                state(CONTROL_LOST);
+            }
+        }
+        else
+        {
+            state(new_state);
+        }
```

Comment:
Changing transitionToControlling to return a State would dramatically improve this. That way you could do:
```
auto resulting_state = transitionToControlling(new_state)
state(resulting_state)
```

Reply:
Could be. The question is: the state transition should be done completely in the updateHook() or in different parts of the code?

Reply:
It's better to centralize it on the `updateHook`, but this wouldnt change that. You would only output the resulting state from transitionToControlling, while the transition would still happen in the `updateHook`.

Reply:
I can see the point of leaving it, as it makes the logic explicit. I think this is too wordy, so that's a tradeoff. I guess clarity trumps cleanliness.